datasets/dk/pep/crawler.py

```diff
@@ -0,0 +1,191 @@
+
+    Args:
+        sheet (openpyxl.worksheet.worksheet.Worksheet): The sheet to parse.
+
+    Yields:
+        Generator[dict, None, None]: Each row of the table as a dictionary.
+    """
+
+    headers = None
+    for idx, row in enumerate(sheet.rows):
+        cells = [c.value for c in row]
+        if headers is None:
+            headers = header_names(cells)
+            continue
+        record = {}
+        for header, value in zip(headers, cells):
+            if isinstance(value, datetime):
+                value = value.date()
+            value = stringify(value)
+            if value is not None:
+                record[header] = value
+        if len(record) == 0:
+            continue
+        yield record
+
+
+def parse_current_pep(
+    sheet: openpyxl.worksheet.worksheet.Worksheet, context: Context
+) -> Generator[dict, None, None]:
+    """This function parses the current PEP list, which is in the format of multiple tables in the first sheet of the Excel file.
+
+    Args:
+        sheet (openpyxl.worksheet.worksheet.Worksheet): The sheet to parse.
+
+    Yields:
+        Generator[dict, None, None]: Each row of the table as a dictionary.
+    """
+    for idx, row in enumerate(sheet.rows):
+        # First row is just the date and name of the dataset
+        # Second row is the header
+        if idx in [0, 1]:
+            continue
+
+        # Every row can be either:
+        # - Name of the organization
+        # - Data of the PEP
+
+        # If there are no values in the row, then it's an empty row
+        if all([c.value is None for c in row]):
+            pass
+
+        # If there is only one not null value in the row and it's the first, then it's the name of the list
+        elif sum([c.value is not None for c in row]) == 1 and row[0].value is not None:
+            list_name = row[0].value
+
+        # If at least one of the 2nd, 3rd, 4th, 5th and 6th columns have values, then it's a PEP data
+        elif any([c.value is not None for c in row[1:6]]):
+            # Meaning "No boards"
+            if row[1].value == "Ingen styrelser etc.":
+                continue
+
+            yield {
+                "last-name": row[1].value,
+                "first-name": row[2].value,
+                "position": row[3].value,
+                "birth-date": row[4].value,
+                "start-date": row[5].value,
+                "list-name": list_name,
+            }
+
+        else:
+            context.log.warning(f"Couldn't parse row: {[c.value for c in row]}")
+
+
+def crawl_current_pep_item(input_dict: dict, context: Context):
+    first_name = input_dict.pop("first-name")
+    last_name = input_dict.pop("last-name")
+
+    entity = context.make("Person")
+    entity.id = context.make_slug(first_name, last_name)
+    h.apply_name(entity, first_name=first_name, last_name=last_name)
+
+    entity.add(
+        "birthDate", h.parse_date(input_dict.pop("birth-date"), formats=["%d.%m.%Y"])
```

Comment:
We're doing a few of these parse calls with this format, perhaps extract into a function?